wagtail/snippets/tests/test_snippets.py

```diff
@@ -616,6 +639,99 @@ def test_locale_selector_not_present_on_non_translatable_snippet(self):
         self.assertNotContains(response, "Switch locales")
 
 
+class TestCreateDraftStateSnippet(TestCase, WagtailTestUtils):
+    def setUp(self):
+        self.user = self.login()
+
+    def get(self):
+        return self.client.get(reverse("wagtailsnippets_tests_draftstatemodel:add"))
+
+    def post(self, post_data={}):
+        return self.client.post(
+            reverse("wagtailsnippets_tests_draftstatemodel:add"),
+            post_data,
+        )
+
+    def test_get(self):
+        response = self.get()
+
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, "wagtailsnippets/snippets/create.html")
+
+        # The save button should be labelled "Save draft"
+        self.assertContains(response, "Save draft")
+        # The publish button should exist
+        self.assertContains(response, "Publish")
+        # The publish button should have name="action-publish"
+        self.assertContains(
+            response,
+            '<button type="submit" name="action-publish" value="action-publish" class="button action-save button-longrunning" data-clicked-text="Publishing…">',
+        )
+        # The status side panel should not be shown
+        self.assertNotContains(
+            response,
+            '<div class="form-side__panel" data-side-panel="status">',
+        )
+
+    def test_save_draft(self):
+        response = self.post(post_data={"text": "Draft-enabled Foo"})
+        snippet = DraftStateModel.objects.get(text="Draft-enabled Foo")
+
+        self.assertRedirects(
+            response, reverse("wagtailsnippets_tests_draftstatemodel:list")
+        )
+
+        # The instance should be created
+        self.assertEqual(snippet.text, "Draft-enabled Foo")
+
+        # The instance should be a draft
+        self.assertIs(snippet.live, False)
```

Comment:
We should use ``assertTrue``/``assertFalse`` when comparing directly against ``True`` or ``False``

Reply:
Okay, but in case you didn't know, [`assertTrue`/`assertFalse` tests for truthy/falsy values](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertTrue), so if `snippet.live` is something like `"some truthy value"`, `assertTrue` will pass.